### PR TITLE
Remove Unneeded Reference to repository resource

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Most of the repository references are no longer needed.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -14,23 +14,18 @@ parameters:
...
     Windows_Python27:
-      OSName: 'Windows'
       OSVmImage: 'windows-2019'
       PythonVersion: '2.7'
     Linux_PyPy3:
-      OSName: 'Linux'
       OSVmImage: 'ubuntu-18.04'
       PythonVersion: 'pypy3'
     Linux_Python38:
-      OSName: 'Linux'
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:33]: While this is good clean-up was it actually needed for the other work?

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 2]
```diff
@@ -1,12 +1,38 @@
+resources:
+  repositories:
+    - repository: azure-sdk-tools
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml:3]: As a follow-up item for all the language repo's please figure out all the places we are using these repository references and see if we can eliminate them. That goes for both azure-sdk-tools and azure-sdk-build-tools. I suspect azure-sdk-build-tools should only ever be needed for java and .NET.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml (hunk 1 above)
Comment (chidozieononiwu) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:33]: We decided to remove this, and instead create the variable after the verifyagentos script has run. This is the pattern we follow in other places so I updated it here. The variable is only used later to name the test result.

**Pull request merged**